pipeline/config.go

```diff
@@ -53,12 +59,27 @@ func (c Config) BuildPipeline(bc operator.BuildContext, defaultOperator operator
 	return NewDirectedPipeline(operators)
 }
 
-func getBuildContextWithDefaultOutput(configs []operator.Config, i int, bc operator.BuildContext) operator.BuildContext {
-	if i+1 >= len(configs) {
-		return bc
+func SetOutputIDs(operators []operator.Operator, pluginMap map[string]string) error {
+	for i, op := range operators {
+		if len(op.GetOutputIDs()) == 0 && i+1 < len(operators) {
+			err := op.SetOutputIDs([]string{operators[i+1].ID()})
+			if err != nil {
+				return err
+			}
+			continue
+		}
+		allOutputs := []string{}
+		for _, id := range op.GetOutputIDs() {
```

Comment:
Can you clarify what this loop is doing?

Reply:
it checks if there are any plugins within the `outputIDs` of the operator. If there is, it changes the output to be the first op in the plugin.